### PR TITLE
Migrating AppError to error for mailservice

### PR DESCRIPTION
**Pull request by jespino**
#### Summary
Migrating AppError to error for mailservice

#### Ticket Link
[MM-32480](https://mattermost.atlassian.net/browse/MM-32480)

#### Related PRs
enterprise: mattermost/enterprise#911

#### Release Note
```release-note
NONE
```

### CODE REVIEW

**Review by reflog — APPROVED** [review]
LGTM

File: app/server.go [hunk 1]
```diff
@@ -469,8 +469,8 @@ func NewServer(options ...Option) (*Server, error) {
 	}
 	s.WebSocketRouter.app = fakeApp
 
-	if appErr := mailservice.TestConnection(s.Config()); appErr != nil {
-		mlog.Error("Mail server connection test is failed: " + appErr.Message)
+	if nErr := mailservice.TestConnection(s.Config()); nErr != nil {
+		mlog.Error("Mail server connection test is failed: " + nErr.Error())
```
Comment (agnivade) [inline app/server.go:473]: We can use `mlog.Err(nErr)`.

File: services/mailservice/mail.go [hunk 2]
```diff
@@ -129,19 +128,19 @@ func ConnectToSMTPServerAdvanced(connectionInfo *SmtpConnectionInfo) (net.Conn,
 
 		conn, err = tls.DialWithDialer(dialer, "tcp", smtpAddress, tlsconfig)
 		if err != nil {
-			return nil, model.NewAppError("SendMail", "utils.mail.connect_smtp.open_tls.app_error", nil, err.Error(), http.StatusInternalServerError)
+			return nil, errors.Wrap(err, "unable to connect to the smtp server through tls")
```
Comment (agnivade) [inline services/mailservice/mail.go:131]: SMTP, TLS

File: services/mailservice/mail.go [hunk 3]
```diff
@@ -129,19 +128,19 @@ func ConnectToSMTPServerAdvanced(connectionInfo *SmtpConnectionInfo) (net.Conn,
...
 		}
 	} else {
 		conn, err = dialer.Dial("tcp", smtpAddress)
 		if err != nil {
-			return nil, model.NewAppError("SendMail", "utils.mail.connect_smtp.open.app_error", nil, err.Error(), http.StatusInternalServerError)
+			return nil, errors.Wrap(err, "unable to connect to the smtp server")
```
Comment (agnivade) [inline services/mailservice/mail.go:136]: SMTP

File: services/mailservice/mail.go [hunk 4]
```diff
@@ -174,16 +173,16 @@ func NewSMTPClientAdvanced(ctx context.Context, conn net.Conn, hostname string,
 	case <-ctx.Done():
 		err := ctx.Err()
 		if err != nil && err.Error() != "context canceled" {
-			return nil, model.NewAppError("SendMail", "utils.mail.connect_smtp.open_tls.app_error", nil, err.Error(), http.StatusInternalServerError)
+			return nil, errors.Wrap(err, "unable to connect to the smtp server")
```
Comment (agnivade) [inline services/mailservice/mail.go:176]: SMTP

File: services/mailservice/mail.go [hunk 5]
```diff
@@ -222,14 +221,14 @@ func NewSMTPClient(ctx context.Context, conn net.Conn, config *model.Config) (*s
 	)
 }
 
-func TestConnection(config *model.Config) *model.AppError {
+func TestConnection(config *model.Config) error {
 	if !*config.EmailSettings.SendEmailNotifications {
-		return &model.AppError{Message: "SendEmailNotifications is not true"}
+		return errors.New("disabled notifications")
```
Comment (agnivade) [inline services/mailservice/mail.go:226]: I think the previous text was more lucid: "SendEmailNotifications is not true"

File: services/mailservice/mail.go [hunk 6]
```diff
@@ -349,37 +348,37 @@ func SendMail(c smtpClient, mail mailData, fileBackend filesstore.FileBackend, d
...
 
 	if err = c.Rcpt(mail.smtpTo); err != nil {
-		return model.NewAppError("SendMail", "utils.mail.send_mail.to_address.app_error", nil, err.Error(), http.StatusInternalServerError)
+		return errors.Wrap(err, "unable to set the to address")
 	}
 
 	w, err := c.Data()
 	if err != nil {
-		return model.NewAppError("SendMail", "utils.mail.send_mail.msg_data.app_error", nil, err.Error(), http.StatusInternalServerError)
+		return errors.Wrap(err, "unable to set the message writer")
```
Comment (agnivade) [inline services/mailservice/mail.go:372]: failed to add email message data

File: services/mailservice/mail.go [hunk 7]
```diff
@@ -349,37 +348,37 @@ func SendMail(c smtpClient, mail mailData, fileBackend filesstore.FileBackend, d
...
 
 	_, err = m.WriteTo(w)
 	if err != nil {
-		return model.NewAppError("SendMail", "utils.mail.send_mail.msg.app_error", nil, err.Error(), http.StatusInternalServerError)
+		return errors.Wrap(err, "unable to set the write the message to the message writer")
 	}
 	err = w.Close()
 	if err != nil {
-		return model.NewAppError("SendMail", "utils.mail.send_mail.close.app_error", nil, err.Error(), http.StatusInternalServerError)
+		return errors.Wrap(err, "unable to set the close the message writer")
```
Comment (agnivade) [inline services/mailservice/mail.go:381]: failed to close connection to the SMTP server

File: services/mailservice/mail.go [hunk 8]
```diff
@@ -349,37 +348,37 @@ func SendMail(c smtpClient, mail mailData, fileBackend filesstore.FileBackend, d
...
 	w, err := c.Data()
 	if err != nil {
-		return model.NewAppError("SendMail", "utils.mail.send_mail.msg_data.app_error", nil, err.Error(), http.StatusInternalServerError)
+		return errors.Wrap(err, "unable to set the message writer")
 	}
 
 	_, err = m.WriteTo(w)
 	if err != nil {
-		return model.NewAppError("SendMail", "utils.mail.send_mail.msg.app_error", nil, err.Error(), http.StatusInternalServerError)
+		return errors.Wrap(err, "unable to set the write the message to the message writer")
```
Comment (agnivade) [inline services/mailservice/mail.go:377]: failed to write the email message

**Review by agnivade — APPROVED** [review]

**Comment by jespino** [comment]
@lindalumitchell This affects all the email sending. I think it is just fine to include this as part of the RC testing.

**Review by lindalumitchell — APPROVED** [review]

**Pull request merged**